packages/automated_actions_utils/automated_actions_utils/aws_api.py

```diff
@@ -0,0 +1,138 @@
+    from mypy_boto3_rds.type_defs import EventTypeDef
+
+
+import logging
+
+from automated_actions.config import settings
+
+from automated_actions_utils.vault_client import (
+    SecretFieldNotFoundError,
+    VaultClient,
+    VaultSecret,
+)
+
+log = logging.getLogger(__name__)
+
+
+class AWSCredentials(ABC):
+    @abstractmethod
+    def build_session(self) -> Session:
+        """Builds and returns a boto3 Session using the implemented credentials."""
+
+
+class AWSStaticCredentials(BaseModel, AWSCredentials):
+    """Represents static AWS credentials consisting of an access key ID, secret access key, and region."""
+
+    access_key_id: str
+    secret_access_key: str
+    region: str
+
+    def build_session(self) -> Session:
+        """Builds and returns a boto3 Session using static credentials."""
+        return Session(
+            aws_access_key_id=self.access_key_id,
+            aws_secret_access_key=self.secret_access_key,
+            region_name=self.region,
+        )
+
+
+def get_aws_credentials(vault_secret: VaultSecret, region: str) -> AWSCredentials:
+    """Retrieves AWS credentials from Vault and returns them as an AWSCredentials object.
+
+    Args:
+        vault_secret: The VaultSecret object containing path and version information.
+        region: The AWS region for the credentials.
+
+    Returns:
+        An AWSStaticCredentials object populated with credentials from Vault.
+
+    Raises:
+        SecretFieldNotFoundError: If 'aws_access_key_id' or 'aws_secret_access_key'
+                                  are not found in the secret.
+    """
+    vault_client = VaultClient(
+        server_url=settings.vault_server_url,
+        role_id=settings.vault_role_id,
+        secret_id=settings.vault_secret_id,
+        kube_auth_role=settings.vault_kube_auth_role,
+        kube_auth_mount=settings.vault_kube_auth_mount,
+    )
+
+    secret = vault_client.read_secret(
+        path=vault_secret.path,
+        version=str(vault_secret.version),
+    )
+
+    if "aws_access_key_id" not in secret or "aws_secret_access_key" not in secret:
+        raise SecretFieldNotFoundError(
+            f"aws_access_key_id or aws_secret_access_key not found in secret {vault_secret.path}"
+        )
+
+    return AWSStaticCredentials(
+        access_key_id=secret["aws_access_key_id"],
+        secret_access_key=secret["aws_secret_access_key"],
+        region=region,
+    )
+
+
+class AWSApi:
+    """A client for interacting with AWS services."""
+
+    def __init__(self, credentials: AWSCredentials, region: str | None) -> None:
+        self.session = credentials.build_session()
+        self.config = Config(
+            region_name=region,
+            retries={
+                "max_attempts": 5,
+                "mode": "standard",
+            },
+        )
+
+    @property
+    def rds_client(self) -> "RDSClient":
+        """Gets a boto3 RDS client."""
+        return self.session.client("rds", config=self.config)
```

Comment:
please persist `rds_client` on `__init__` or use `cached_property` so the same client object can be reused.

Reply:
Changed in [516b0fc](https://github.com/app-sre/automated-actions/pull/95/commits/516b0fc55eb2190be9eb77e9336746b8290345e6)